mycroft/skills/core.py

```diff
@@ -275,11 +277,23 @@ def settings(self):
             self._settings = SkillSettings(self._dir, self.name)
             return self._settings
 
-    def bind(self, emitter):
-        """ Register emitter with skill. """
-        if emitter:
-            self.emitter = emitter
-            self.enclosure = EnclosureAPI(emitter, self.name)
+    @property
+    def emitter(self):
+        """ Backwards compatibility. """
+        LOG.info('self.emitter is depreciated, use self.messagebus instead')
+        return self.messagebus
+
+    @emitter.setter
+    def emitter(self, value):
+        """ Backwards compatibility. """
+        LOG.info('self.emitter is depreciated, use self.messagebus instead')
+        self.messagebus = value
```

Comment:
in this case wouldn't be self.mesagebus.emit(value) ?

Reply:
This is to fake the set operation

```python

skill.emitter = test_emitter
```

Would throw an exception if the setter method didn't exist. This makes it so that anything that tries to set `.emitter` actually sets `.messagebus`